internal/printer/print-to-tsx.go

```diff
@@ -64,6 +64,10 @@ func getTextType(n *astro.Node) TextType {
 func renderTsx(p *printer, n *Node) {
 	// Root of the document, print all children
 	if n.Type == DocumentNode {
+		props := js_scanner.GetPropsType([]byte(p.sourcetext))
+		if props.Ident != "Record<string, any>" {
+			p.printf(`declare const Astro: import('astro').AstroGlobal<%s>`, props.Ident)
```

Comment:
Not doing a suggestion since I figure you don't necessarily want it in the string itself, but the JSDoc comment for the Astro global itself needs to be there, that one we can't work around unfortunately (as far as I know, at least):

```
/**
 * Astro global available in all contexts in .astro files
 *
 * [Astro documentation](https://docs.astro.build/reference/api-reference/#astro-global)
*/
declare const Astro ...
 ```